src/Symfony/Component/EventDispatcher/EventListener.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+/**
+ * Helper class for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+abstract class EventListener implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public static function getSubscribedEvents(): array
```

Comment:
this should not be `final` to me (if we were to merge this, but I'm objecting, so this is just for the record :) )

Reply:
not marking it `final` would defeat the design and purpose of the class.

Reply:
why so? you said because it would make the method show up in the phpstorm suggestion, are there any other reasons?
to me (as I wrote in other comments) the trait should not provide any other extensibility mean.
There is only one thing to *learn* from Symfony code: the interface and thus the method. Then, there is this helper trait, which removes some boilerplate for common needs.
But for any not-common need, there should be only one way: the interface+method.

Reply:
> why so? you said because it would make the method show up in the phpstorm suggestion, are there any other reasons?

There is no other reasons, it's all about DX in my side.